app/src/main/java/com/randomappsinc/studentpicker/grouping/GroupingMakingFragment.java

```diff
@@ -70,27 +79,62 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle sa
     public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         super.onActivityCreated(savedInstanceState);
         settings = new GroupingSettings(
+                listInfo.getNumNames(),
                 getResources().getInteger(R.integer.default_number_of_names_per_group),
                 getResources().getInteger(R.integer.default_number_of_groups));
         settingsDialog = new GroupingSettingsDialog(getActivity(), settings);
     }
 
+    @Override
+    public void onNameAdded(String name, int amount, String listName) {
+        listInfo = new DataSource(getContext()).getListInfo(listName);
+        settings.setListSize(listInfo.getNumNames());
+        settingsDialog.refreshSetting();
+    }
+
+    @Override
+    public void onNameDeleted(String name, int amount, String listName) {
+        listInfo = new DataSource(getContext()).getListInfo(listName);
```

Comment:
Instead of fetching from DataSource, we can just do what NameAdapter does (check out the code) and mutate the ListInfo we have. This will save a DB read, which is pricey.